File: array/26.py
#!/usr/bin/python
#coding=utf-8
import math

class Solution(object):
    def removeDuplicates(self, nums):
      if not nums:
        return 0
      k=0;
      for i in range(1, len(nums)):
      	if nums[i] != nums[k]:
      		k += 1 
      		#k +=1 before nums[k] = nums[i] because num[k+1] replaced by new value
      		#python does not support ++,not need ;
      		nums[k] = nums[i]
     
      return k+1

if __name__ == "__main__":
	so = Solution()
	nums = [1,1,1,2,2,3]
	print so.removeDuplicates(nums)
	print nums

File: array/26.c
#include<stdio.h>

int removeDuplicates(int* nums, int numsSize) {
  int i,j,k=0;
  
  while (i<numsSize){
      while(nums[i+1] == nums[i])
        i++;
      i++;
      k++;
      nums[k-1]=nums[i-1];
      
  }
  nums[k]='\0';
  return nums;
}

int main(){
  int nums[6] = {1,1,1,2,2,3};
  int a = removeDuplicates(nums,6);
  for(int i=0;i<3;i++)
    printf("%d",nums[i]);
}